### PR TITLE
Fix sharder/miner/validator rewards bug in eventsDB

### PR DESCRIPTION
**Pull request by iamrz1**
## Fixes
In eventsDB for sharder/miner/validator:
- Mismatching name `rewards` and `reward` for the same field
- Reference to non-existing field `total_service_charge`

Ref:
https://0chain.slack.com/archives/C03788G8N2D/p1652975157823899

### CODE REVIEW

File: code/go/0chain.net/smartcontract/dbs/event/stakepool.go [hunk 1]
```diff
@@ -59,24 +59,24 @@ func (edb *EventDb) rewardProvider(spu dbs.StakePoolReward) error {
 		if err != nil {
 			return err
 		}
-		update.Updates["reward"] = validator.Reward + spu.Reward
-		update.Updates["total_service_charge"] = validator.TotalReward + spu.Reward
+		update.Updates["rewards"] = validator.Rewards + spu.Reward
+		update.Updates["service_charge"] = validator.TotalReward + spu.Reward
```
Comment (ssardana08) [inline code/go/0chain.net/smartcontract/dbs/event/stakepool.go]: it should be total_reward

File: code/go/0chain.net/smartcontract/dbs/event/stakepool.go [hunk 2]
```diff
@@ -59,24 +59,24 @@ func (edb *EventDb) rewardProvider(spu dbs.StakePoolReward) error {
...
 		return edb.updateValidator(*update)
 	case spenum.Miner:
 		miner, err := edb.minerAggregateStats(spu.ProviderId)
 		if err != nil {
 			return err
 		}
-		update.Updates["reward"] = miner.Reward + spu.Reward
-		update.Updates["total_service_charge"] = miner.TotalReward + spu.Reward
+		update.Updates["rewards"] = miner.Rewards + spu.Reward
+		update.Updates["service_charge"] = miner.TotalReward + spu.Reward
```
Comment (ssardana08) [inline code/go/0chain.net/smartcontract/dbs/event/stakepool.go]: total_reward

File: code/go/0chain.net/smartcontract/dbs/event/stakepool.go [hunk 3]
```diff
@@ -59,24 +59,24 @@ func (edb *EventDb) rewardProvider(spu dbs.StakePoolReward) error {
...
 		return edb.updateMiner(*update)
 	case spenum.Sharder:
 		sharder, err := edb.sharderAggregateStats(spu.ProviderId)
 		if err != nil {
 			return err
 		}
-		update.Updates["reward"] = sharder.Reward + spu.Reward
-		update.Updates["total_service_charge"] = sharder.TotalReward + spu.Reward
+		update.Updates["rewards"] = sharder.Rewards + spu.Reward
+		update.Updates["service_charge"] = sharder.TotalReward + spu.Reward
```
Comment (ssardana08) [inline code/go/0chain.net/smartcontract/dbs/event/stakepool.go]: total_reward

File: code/go/0chain.net/smartcontract/dbs/event/validator.go [hunk 4]
```diff
@@ -24,7 +24,7 @@ type Validator struct {
 	NumDelegates   int           `json:"num_delegates"`
 	ServiceCharge  float64       `json:"service_charge"`
 
-	Reward      int64 `json:"reward"`
+	Rewards     int64 `json:"rewards"`
```
Comment (ssardana08) [inline code/go/0chain.net/smartcontract/dbs/event/validator.go:27]: doesn't changing the json fails system tests? please ensure that
if it fails, keep the json as it is for now

**Review by ssardana08 — DISMISSED** [review]

File: code/go/0chain.net/smartcontract/dbs/event/validator.go (hunk 4 above)
Comment (iamrz1) [reply code/go/0chain.net/smartcontract/dbs/event/validator.go:27]: Strangely enough, so far the value seems to be unused by get functions. System tests have also passed.

**Review by dabasov — COMMENTED** [review]
lgtm

**Review by dabasov — APPROVED** [review]

File: code/go/0chain.net/smartcontract/dbs/event/stakepool.go (hunk 3 above)
Comment (iamrz1) [reply code/go/0chain.net/smartcontract/dbs/event/stakepool.go]: done

File: code/go/0chain.net/smartcontract/dbs/event/stakepool.go (hunk 2 above)
Comment (iamrz1) [reply code/go/0chain.net/smartcontract/dbs/event/stakepool.go]: done

File: code/go/0chain.net/smartcontract/dbs/event/stakepool.go (hunk 1 above)
Comment (iamrz1) [reply code/go/0chain.net/smartcontract/dbs/event/stakepool.go]: done

**Comment by iamrz1** [comment]
@dabasov can we have this merged? (I'm not [authorized](https://docs.github.com/articles/about-protected-branches/) to merge pull requests)

**Pull request merged**